File: scala/bob/src/main/scala/bob.scala
class Bob {
  def hey(input : String) = {
    if (isAllCaps(input)) {
      "Whoa, chill out!"
    } else if(isSilence(input)) {
      "Fine. Be that way!"
    } else if(isQuestion(input)) {
      "Sure."
    } else {
      "Whatever."
    }
  }

  private
  def isAllCaps(input: String) : Boolean = {
    hasLetters(input) &&
    input == input.toUpperCase
  }

  def isQuestion(input: String) : Boolean = {
    input.last == '?'
  }

  def hasLetters(input : String) : Boolean = {
    input.matches(".*[a-zA-Z]+.*")
  }

  def isSilence(input : String) : Boolean = {
    input.trim.isEmpty
  }
}

File: erlang/bob/bob.erl
-module(bob).
-export([response_for/1]).

response_for(Message) ->
	IsShout = is_shout(Message),
	IsQuestion = is_question(Message),
	IsSilence = is_silence(Message),
	if
		IsShout -> "Whoa, chill out!";
		IsQuestion -> "Sure.";
		IsSilence -> "Fine. Be that way!";
		true -> "Whatever."
	end.

is_shout(Message) ->
	(string:to_upper(Message) == Message) and (string:to_lower(Message) =/= Message).

is_question(Message) ->
	string:right(Message, 1) == "?".

is_silence(Message) ->
	string:strip(Message) == "".
